internal/controllers/core/cluster/reconciler.go

```diff
@@ -357,6 +377,50 @@ func (r *Reconciler) populateK8sMetadata(ctx context.Context, clusterNN types.Na
 	}
 }
 
+func (r *Reconciler) writeFrozenKubeConfig(ctx context.Context, nn types.NamespacedName, config *api.Config) string {
+	config = config.DeepCopy()
+	err := api.MinifyConfig(config)
+	if err != nil {
+		logger.Get(ctx).Warnf("Minifying Kubernetes config: %v", err)
+		return ""
+	}
+
+	err = api.FlattenConfig(config)
+	if err != nil {
+		logger.Get(ctx).Warnf("Flattening Kubernetes config: %v", err)
+		return ""
+	}
+
+	obj, err := latest.Scheme.ConvertToVersion(config, latest.ExternalVersion)
+	if err != nil {
+		logger.Get(ctx).Warnf("Converting Kubernetes config: %v", err)
+		return ""
+	}
+
+	printer := printers.YAMLPrinter{}
+	path, err := r.base.ConfigFile(filepath.Join("cluster", nn.Name))
```

Comment:
Nit: `RuntimeFile` seems more appropriate for this

Reply:
done